src/unix/linux/cpu.rs

```diff
@@ -830,3 +841,14 @@ pub(crate) fn get_vendor_id_and_brand() -> HashMap<usize, (String, String)> {
     }
     cpus
 }
+
+fn get_cpu_arch() -> CpuArch {
+    let mut s = String::new();
+    if let Err(_e) = File::open("/proc/sys/kernel/arch").and_then(|mut f| f.read_to_string(&mut s))
```

Comment:
This doesn't work (at least on my machine). I was able to get this information in C with:

```c
#include <stdio.h>
#include <sys/utsname.h>

int main() {
  struct utsname osinfo;
  uname(&osinfo);
  printf("%s\n", osinfo.machine);
  return 0;
}
```

Luckily, I already use `uname` in the `kernel_version` method. You can reuse the code and instead get the `machine` field value.